src/Interfaces/MOI_interface.jl

```diff
@@ -689,6 +692,7 @@ function MOI.optimize!(model::Optimizer)
     model.solver = MadNLPSolver(model.nlp; model.options...)
     model.result = solve!(model.solver)
     model.solve_time = model.solver.cnt.total_time
+    model.solve_iterations = model.solver.cnt.k
```

Comment:
How about using a getter instead? Something like `get_solve_iterations(model)`. 
The idea would be to avoid duplicating the information between `model.solve_iterations` and `model.solver.cnt.k`